File: examples/typed-authoring-gradle/src/main/java/laughing/man/commits/examples/typedgradle/Employee.java
package laughing.man.commits.examples.typedgradle;

import laughing.man.commits.annotations.Exclude;
import laughing.man.commits.annotations.GeneratePojoLensTypedFields;

@GeneratePojoLensTypedFields
public class Employee {

    public boolean active;
    public String department;
    public int salary;
    public Status status;
    @Exclude
    public String internalCode;

    public enum Status {
        ACTIVE,
        INACTIVE
    }
}

File: examples/typed-compiler-gradle-kotlin/src/main/kotlin/laughing/man/commits/examples/typedkotlin/Employee.kt
package laughing.man.commits.examples.typedkotlin

import laughing.man.commits.annotations.Exclude
import laughing.man.commits.annotations.GeneratePojoLensTypedFields

@GeneratePojoLensTypedFields
class Employee {
    @JvmField
    var active: Boolean = false

    @JvmField
    var department: String = ""

    @JvmField
    var salary: Int = 0

    @JvmField
    var status: Status = Status.ACTIVE

    @field:Exclude
    @JvmField
    var internalCode: String = ""

    enum class Status {
        ACTIVE,
        INACTIVE
    }
}
